reworked

sadensmol opened the pull request:
Reworked app in according to Unity test app
Added all needed libraries for both platform to get them up to date and tested
Added separate configs for both platforms with different package ids to get all networks working at start!

Commit:
reworked

Commit:
reworked and repacked to get it worked on android device

Commit:
added target apk for android platform for test

ekilah commented on README.md:
```diff
@@ -1,43 +1,66 @@
+AdColony SDK 2.0.9
+AdMob SDK 7.4.1
+AppLovin SDK 1.0.10
+Chartboost IOS SDK 5.5.3
```
here and elsewhere, use `iOS` instead of `IOS`

sadensmol replied:
fixed

ekilah commented on README.md:
```diff
@@ -1,43 +1,66 @@
+Libraries like FacebookAudienceNetwork.ane has internal library android-support-v4.jar 
+
+- Error java.lang.OutOfMemoryError: GC overhead limit exceeded. Increase memory for java in Adobe Air/Flex framework. 
+open jvm.config and set java.args=-Xmx1024m -Xms512m -Dsun.io.useCanonCaches=false
```
path for this config file? also is this error experienced on device or while building?

sadensmol replied:
added path and some explanation

ekilah commented on src/FlexExampleApp.mxml:
```diff
@@ -1,14 +1,30 @@
+
+        protected function onCreationComplete():void {
+
+            HeyzapAds.getInstance().start("9317e0b9c4eaf41ed68e907e22ab3c5a", HeyzapAds.FLAG_NONE);
```
you'll want to keep this as something like `"<APP_ID>"` or `"ENTER YOUR PUBLISHER ID HERE"` (or even `"1234"`) instead of giving what I assume is your own ID in the sample project

ekilah replied:
also, for flags, it might be useful to use the flag to disable automatic prefetching, both as an example of flag usage and so that the fetch button is the only way ads are fetched.

sadensmol replied:
The main idea of this test application to give developers ability to test how heyzap works.
So we have pre setup all package names and this app id. To get it just working without any additional steps for developers. Maybe we can add some other ID or just leave it empty? I tried with empty string and in some cases it didn't work well, so I put my ID there. What do you think?

ekilah commented on src/flex-example-app-android.xml:
```diff
@@ -0,0 +1,94 @@
+        <extensionID>com.heyzap.sdk.extensions.unityads</extensionID>
+        <extensionID>com.heyzap.sdk.extensions.facebook.audiencenetwork</extensionID>
+
+        <!--extensionID>com.pozirk.ads.AdMob</extensionID-->
```
leave a comment as to when you might use this, or leave it out completely?

sadensmol replied:
Removed it. I haven't tested this yet but will do it today. Actually looks that AdMob is working without this additional GPS ANE just with Vungle's one. I have installed Genymotion and will test with it today!

ekilah commented on src/flex-example-app-ios.xml:
```diff
@@ -0,0 +1,58 @@
+
+        <extensionID>com.heyzap.sdk.extensions.facebook.audiencenetwork</extensionID>
+
+<!--
```
why are these commented out?

sadensmol replied:
yes, fixed, sorry my fault.

ekilah commented on src/flex-example-app-ios.xml:
```diff
@@ -0,0 +1,58 @@
+            <!--<key>UIStatusBarStyle</key>-->
+            <!--<string>UIStatusBarStyleBlackOpaque</string>-->
+            <key>MinimumOSVersion</key>
+            <string>8.0</string>
```
is there a reason this only runs on iOS 8 or greater?

sadensmol replied:
App Lovin ANE has 8.0 as min req. Don't know why but actually about 86% of all iDevices have 8.0 and greater version according to official stats. So i think it's okay.

ekilah commented on src/model/AppModel.as:
```diff
@@ -0,0 +1,12 @@
+[Bindable]
+public class AppModel {
+    public var showLogs:Boolean = true;
+    public var bannerIsShown:Boolean;
```
what banner does this refer to? a comment or better name might help here.

ekilah commented on src/model/AppParameters.as:
```diff
@@ -0,0 +1,5 @@
+package {
+public class AppParameters {
```
doesn't seem like this is used anywhere.

sadensmol replied:
removed. Had some ideas on this  file, but it looks that nothing to place there!

ekilah commented on src/views/FlexExampleAppHomeView.mxml:
```diff
@@ -1,204 +1,129 @@
+        }
+
+        private function getEnteredAdTagOrNull():String{
+            if (!adTagTextInput.text) return null;
```
does this return `null` when the tag text field is empty? that'd be the desired behavior.

sadensmol replied:
if (!String) in as3 returns true if text is null or empty. 
http://help.adobe.com/en_US/ActionScript/3.0_ProgrammingAS3/WS5b3ccc516d4fbf351e63e3d118a9b90204-7f87.html

ekilah commented on src/views/FlexExampleAppHomeView.mxml:
```diff
@@ -1,204 +1,129 @@
+
+        private function onShowButtonClick():void {
+            if (currentActionObject instanceof  BannerAd) {
+                currentActionObject.show(BannerAd.POSITION_BOTTOM,getEnteredAdTagOrNull()); //todo add position
```
it'd be useful if there was another radio button to choose the banner location, top or bottom! (as your comment states)

sadensmol replied:
added

ekilah commented on src/views/FlexExampleAppHomeView.mxml:
```diff
@@ -1,204 +1,129 @@
+        private function onShowButtonClick():void {
+            if (currentActionObject instanceof  BannerAd) {
+                currentActionObject.show(BannerAd.POSITION_BOTTOM,getEnteredAdTagOrNull()); //todo add position
+                appModel.bannerIsShown = true;
```
this is not strictly true. banner ads can fail to show, in which case the test app will fall into an inconsistent state. You could listen for the `loaded` callback and set this to true there.

(Or, even better, don't have state. let all the buttons stay enabled, so they strictly make function calls, whether or not they're appropriate calls to make. let the user determine if it's the right time to call those methods, by showing the callbacks in the console view you have)

sadensmol replied:
but we use this only tho change show/hide button state. if banner fails user can just click hide and show again to try again. I think it's just test app, so it might be enough.
But if you want i can add additional listener for BannerAd to change buttons state on fail. Or just add some message to log - like "click hide and show again" if failed.please update

ekilah commented on src/views/FlexExampleAppHomeView.mxml:
```diff
@@ -1,204 +1,129 @@
+            } else if (videoRadioButton.selected) {
+                currentActionObject = VideoAd.getInstance();
+            } else if (bannerRadioButton.selected) {
+                currentActionObject = BannerAd.getInstance();
```
again, state may be an issue here, because it is valid to show a video or interstitial while a banner is also showing. using the same button for showing/hiding a banner as showing another ad is problematic for this reason.

sadensmol replied:
good point. Now when user change selection for current type I added hide function to hide banner before continue and also change buttons state

ekilah commented on src/FlexExampleApp.mxml:
```diff
@@ -1,14 +1,30 @@
+        [Bindable]
+        private var appModel:AppModel = new AppModel();
+
+        protected function onCreationComplete():void {
```
the first thing you should do (before starting the SDK) is enable the debug logs, since the UI defaults to having the logs show as "on", but the SDK does not (necessarily) start with the logs enabled.

sadensmol replied:
Sorry i didn't get this note. Is there any method to enable SDK logs?

ekilah replied:
this goes along with what I realized later, we will add this feature later to the SDK.

ekilah commented on src/views/components/LogConsole.mxml:
```diff
@@ -0,0 +1,163 @@
+
+            switch (event.type) {
+                case HeyzapEvent.IS_AVAILABLE:
+                    if (IncentivizedAd.getInstance().isAvailable()) {
```
the `IS_AVAILABLE` and `FETCH_FAILED` events are opposites, and are called in response to `fetch`es; you should not have to check `IncentivizedAd.getInstance().isAvailable()` here, as it should always be true (although since you're not passing a tag in this check, it wouldn't necessarily always be true).

sadensmol replied:
removed

ekilah commented on src/views/FlexExampleAppHomeView.mxml:
```diff
@@ -1,204 +1,129 @@
+                currentActionObject.show(BannerAd.POSITION_BOTTOM,getEnteredAdTagOrNull()); //todo add position
+                appModel.bannerIsShown = true;
+            } else {
+                currentActionObject.show(getEnteredAdTagOrNull());
```
another `if` case here in the future could be for incentivized ads - similar to [the unity test app](https://github.com/Heyzap/unity-example-app/blob/master/Assets/Scripts/AdManager.cs#L132), we can demonstrate the use of the incentivizedInfo. (but since this feature isn't released yet this will have to come later)

sadensmol replied:
I will add this function

Commit:
uncommented ANEs for ios build

Commit:
reworked libs for ios platform. Put back original facebook ane only for ios, moved rapacked one to android libs

ekilah commented on src/flex-example-app-android.xml:
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<application xmlns="http://ns.adobe.com/air/application/19.0">
+    <id>com.example.android.snake</id>
```
we'll want to use a different default bundle ID and have the README suggest changing this. similar to the app ID

sadensmol replied:
okay, i will add some instructions on how to change code to get it worked with users accounts
but the main idea to show developers that all stuff is working and then ask them to change parameters if they want to test their ads.

ekilah replied:
but we also want them to connect it to their account so that they can control their account from our website. they won't have access to the dashboard site for our own sample apps

ekilah commented on src/flex-example-app-ios.xml:
```diff
@@ -0,0 +1,56 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<application xmlns="http://ns.adobe.com/air/application/19.0">
+    <id>com.EnterpriseHeyzap.HeyzapSDKTest</id>
```
same here as with the android bundle ID

sadensmol replied:
okay, i will add some instructions on how to change code to get it worked with users accounts
but the main idea to show developers that all stuff is working and then ask them to change parameters if they want to test their ads.

ekilah commented on src/views/FlexExampleAppHomeView.mxml:
```diff
@@ -1,204 +1,129 @@
+        }
+
+        private function onShowButtonClick():void {
+            if (currentActionObject instanceof  BannerAd) {
```
intellij says `instanceof` is deprecated in favor of `is` - does it make a difference to update this and avoid the warning?

sadensmol replied:
changed!

ekilah commented on src/views/components/LogConsole.mxml:
```diff
@@ -0,0 +1,163 @@
+        }
+
+        public function showLine(string:String):void {
+            logLines.addItem(string);
```
some sort of timestamp might be nice here for context, it's been useful before in our other test apps

sadensmol replied:
there is no much space for that. but i will add

ekilah commented on src/model/AppModel.as:
```diff
@@ -0,0 +1,12 @@
+package model {
+[Bindable]
+public class AppModel {
+    public var showLogs:Boolean = true;
```
doesn't seem like this value is used to change the logging behavior

ekilah replied:
OHHH I see. you're using it to enable/disable the log view. That's not so useful.

The checkbox should call a method (which does not exist in the ANE yet, as I now see) that enables/disables verbose logging (i.e.: to `adb` or the iOS device logs). https://github.com/Heyzap/unity-example-app/blob/master/Assets/Scripts/AdManager.cs#L189 is an example. There is a ticket to include this in the ANE, so for now you can comment this out of the test app.

sadensmol replied:
I've disabled it for now. Will restore back when task with verbose logging will be done.

Commit:
adds to gitignore and adds missing activity for heyzap exchange

Commit:
added customInfo fon incentivised ads
added readme on how to get started with own id
added timestamp to logs
clean up and some fixes according to PR comments

Commit:
Merge branch 'app_rework_step1' of https://github.com/Heyzap/flex-example-app into app_rework_step1

Commit:
removed default package ids, APP_ID
reworked banners to work along with other ads.
added new notes to read.me
clean up code

Commit:
making denis' PR README pretty
